### PR TITLE
Fix/logging

### LINKED ISSUE #663: ooniprobe should delete old reports when system is running low on disk space
`ooniprobe version: 2.0.0`
`lepidopter version: 0.3.4-beta`
`lepidopter-update version: 3`

ooniprobe measurements and log directories total file size:
```
1.4G    /var/log/ooni
1006M   /var/lib/ooni/measurements/
```

Disk space of lepidopter root (/):
```
$ df -h /
Filesystem      Size  Used Avail Use% Mounted on
/dev/root       3.6G  3.1G  220M  94% /
```

/var/log/ooni/ooniprobe.log excerpt:
```
2016-10-31 18:00:02,378733+0000 [-] Running task delete-old-reports
2016-10-31 18:00:02,384271+0000 [-] Did not run delete-old-reports
2016-10-31 18:00:02,385393+0000 [-] Ran delete-old-reports
2016-10-31 18:00:02,386288+0000 [-] Running task check-measurement-quota
2016-10-31 18:00:02,415338+0000 [-] You are about to reach the maximum allowed quota. Be careful
2016-10-31 18:00:02,418929+0000 [-] Ran check-measurement-quota
```
# Please explain the issue. The first line will become the title. Trailing
# lines starting with '#' (like these) will be ignored, and empty messages will
# not be submitted. Issues are formatted with GitHub Flavored Markdown (GFM):
#
#   http://github.github.com/github-flavored-markdown
#
```

### PR DESCRIPTION
**Pull request by hellais**
This fixes #663

### CODE REVIEW

File: ooni/settings.py [hunk 1]
```diff
@@ -22,6 +22,13 @@
     # Where OONIProbe should be writing its log file
     # logfile: {logfile}
     # loglevel: WARNING
+    # Will rotate logs daily when "daily" or based on rotate_length
+    # when "length" is set. null to disable rotation.
+    # rotate: daily
+    # rotate_length: 1M
+    # Sets an upper bound on the number of rotated files, only works when
```
Comment (bassosimone) [inline ooni/settings.py]: This sentence seems truncated

File: data/ooniprobe.conf.sample [hunk 2]
```diff
@@ -6,6 +6,7 @@ basic:
     # Where OONIProbe should be writing its log file
     logfile: /var/log/ooniprobe.log
     loglevel: WARNING
+    logrotate: true
```
Comment (darkk) [inline data/ooniprobe.conf.sample]: s/logrotate: true/rotate: 'daily'/

File: ooni/utils/log.py [hunk 3]
```diff
@@ -101,7 +102,7 @@ def emit(self, eventDict):
             self.write(text + "\n")
             self.flush()
 
-class MsecLogObserver(tw_log.FileLogObserver):
+class MsecLogObserver(LogLevelObserver):
```
Comment (bassosimone) [inline ooni/utils/log.py]: Why this?

File: ooni/geoip.py [hunk 4]
```diff
@@ -164,8 +164,10 @@ class ProbeIP(object):
 
     def __init__(self):
         self.geoIPServices = {
-            'ubuntu': UbuntuGeoIP,
-            'duckduckgo': DuckDuckGoGeoIP
+            'ubuntu': UbuntuGeoIP
+            # We are disabling this because it sometimes creates parsing
+            # errors.
+            # 'duckduckgo': DuckDuckGoGeoIP
```
Comment (darkk) [inline ooni/geoip.py]: Needs xref to an issue where we fix lack of "redundancy" for the check.

Here is my `myip` script with some samples:
```
# Google Global Cache Installation and Operations Guide / February 17, 2015
# https://peering.google.com/static/downloads/GGCInstallation-R720.pdf
dig @ns1.google.com. -t txt o-o.myaddr.l.google.com. +short | tr -d '"'
dig @ze.akamaitech.net. -t A whoami.akamai.net. +short
# resolver1.opendns.com & resolver2.opendns.com
dig @208.67.222.222 -t A myip.opendns.com. +short
{ curl -sS -4 https://diagnostic.opendns.com/myip; echo; } | cat
{ curl -sS -4 https://yandex.com/internet/api/v0/ip; echo; } | tr -d '"'
```

File: ooni/utils/log.py (hunk 3 above)
Comment (hellais) [inline ooni/utils/log.py]: Not inheriting the `LogLevelObserver` will mean log levels are not respected and everything (even low priority logs) are written to a file.

File: data/ooniprobe.conf.sample (hunk 2 above)
Comment (hellais) [inline data/ooniprobe.conf.sample]: Right, forgot to update the sample config file.

File: ooni/utils/log.py (hunk 3 above)
Comment (bassosimone) [inline ooni/utils/log.py]: Got it!

File: ooni/settings.py [hunk 5]
```diff
@@ -22,6 +22,14 @@
     # Where OONIProbe should be writing its log file
     # logfile: {logfile}
     # loglevel: WARNING
+    # Will rotate logs daily when "daily" or based on rotate_length
+    #  when "length" is set. null to disable rotation.
+    # rotate: daily
+    # rotate_length: 1M
+    # rotate: length
```
Comment (bassosimone) [inline ooni/settings.py]: I think you can avoid this line because you already have `rotate: daily` above

File: data/ooniprobe.conf.sample [hunk 6]
```diff
@@ -4,8 +4,19 @@
...
+    # logfile: {logfile}
+    # loglevel: WARNING
+    # Will rotate logs daily when "daily" or based on rotate_length
+    #  when "length" is set. null to disable rotation.
+    # rotate: daily
+    # rotate_length: 1M
+    # rotate: length
+    # Sets an upper bound on the number of rotated files, only works when
+    #  "length" is the rotation mode
+    # max_rotated_files: None
```
Comment (darkk) [inline data/ooniprobe.conf.sample]: It's `null` in YAML, not `None`.

I'm unsure if oo.conf is parsed with yaml parser or not, but it looks like YAML.

File: ooni/tests/test_utils.py [hunk 7]
```diff
@@ -77,3 +78,49 @@ def test_directory_usage(self):
...
+        mock_config.basic.rotate = 'length'
+        mock_config.basic.rotate_length = '1M'
+
+        ooni_logger = log.OONILogger()
+        ooni_logger.start(logfile=self.path)
+
+        ooni_logger.debug("1"*(1024**2))
+
+        # This will trigger the rotation
+        ooni_logger.debug("2")
```
Comment (bassosimone) [inline ooni/tests/test_utils.py]: Perhaps stupid question, but: you set `advanced.debug` to `False` above... so, why is the `debug()` function then sending logs to the logfile?

Comment (bassosimone) [inline ooni/tests/test_utils.py]: Oh, gosh, this is really confusing!

**Comment by hellais** [comment]
Green light? Merge?

**Comment by darkk** [comment]
LGTM

**Pull request merged**